components/scream/src/physics/p3/p3_inputs_initializer.cpp

```diff
@@ -45,88 +47,251 @@ add_field (const field_type &f, const field_type& f_ref,
 // =========================================================================================
 void P3InputsInitializer::initialize_fields ()
 {
+  using namespace p3;
+  using P3F             = Functions<Real, DefaultDevice>;
+  using Spack           = typename P3F::Spack;
+  using Pack            = ekat::Pack<Real,Spack::n>;
+
   // Safety check: if we're asked to init anything at all,
-  // then we should have been asked to init 10 fields.
+  // To simplify the initializer we first define all the fields we expect to have to initialize.
+  std::vector<std::string> fields_to_init;
+  fields_to_init.push_back("T_atm");
+  fields_to_init.push_back("ast");
+  fields_to_init.push_back("ni_activated");
+  fields_to_init.push_back("nc_nuceat_tend");
+  fields_to_init.push_back("pmid");
+  fields_to_init.push_back("dp");
+  fields_to_init.push_back("zi");
+  fields_to_init.push_back("qv_prev");
+  fields_to_init.push_back("T_prev");
+  fields_to_init.push_back("qv");
+  fields_to_init.push_back("qc");
+  fields_to_init.push_back("qr");
+  fields_to_init.push_back("qi");
+  fields_to_init.push_back("qm");
+  fields_to_init.push_back("nc");
+  fields_to_init.push_back("nr");
+  fields_to_init.push_back("ni");
+  fields_to_init.push_back("bm");
+  fields_to_init.push_back("nccn_prescribed");
+  fields_to_init.push_back("inv_qc_relvar");
+  // TODO: Delete eventually, should instead be set in run interface: 
+  fields_to_init.push_back("th_atm");  
+  fields_to_init.push_back("dz");  
+  fields_to_init.push_back("exner");  
+  fields_to_init.push_back("cld_frac_l");  
+  fields_to_init.push_back("cld_frac_i");  
+  fields_to_init.push_back("cld_frac_r");  
+  // then we should have been asked to init 20 + 6 (to be deleted) fields.
   int count = 0;
-  count += m_fields.count("q");
-  count += m_fields.count("T");
-  count += m_fields.count("ast");
-  count += m_fields.count("ni_activated");
-  count += m_fields.count("nc_nuceat_tend");
-  count += m_fields.count("pmid");
-  count += m_fields.count("dp");
-  count += m_fields.count("zi");
-  count += m_fields.count("qv_prev");
-  count += m_fields.count("T_prev");
+  std::string list_of_fields = "";
+  for (auto name : fields_to_init)
+  {
+    list_of_fields += name;
+    list_of_fields += ", ";
+    count += m_fields.count(name);
+  }
   
   if (count==0) {
     return;
```

Comment:
Just curious--is this not an error condition? It's true that no initialization has to be done in this case, but is it "valid" to call this method under such conditions?

Reply:
So I should switch this to EKAT_REQUIRE?

Reply:
Sure, that works. We can always revisit if it turns out we're wrong about this.